src/creature_states.c

```diff
@@ -2955,7 +2955,77 @@ short creature_wants_salary(struct Thing *creatng)
 
 long setup_head_for_empty_treasure_space(struct Thing *thing, struct Room *room)
 {
-    return _DK_setup_head_for_empty_treasure_space(thing, room);
+    // return _DK_setup_head_for_empty_treasure_space(thing, room);
+    SlabCodedCoords start_slbnum = room->slabs_list;
+   
+    //Find a random slab to start out with
+    unsigned long k;
+    long n = ACTION_RANDOM(room->slabs_count);
+    for (k = n; k > 0; k--)
+    {
+        if (start_slbnum == 0)
+            break;
+        start_slbnum = get_next_slab_number_in_room(start_slbnum);
+    }
+    if (start_slbnum == 0) {
+        ERRORLOG("Taking random slab (%d/%d) in %s index %d failed - internal inconsistency.", (int)n, (int)room->slabs_count, room_code_name(room->kind), (int)room->index);
```

Comment:
Why the int casts?
btw for unsigned numbers you may use `%u`.

Reply:
I don't know, I took it like this from another function.